src/Authoring/WinRT.SourceGenerator/WinRTTypeWriter.cs

```diff
@@ -1561,6 +1561,10 @@ private void AddOverloadAttributeForInterfaceMethods(TypeDeclaration interfaceTy
                         continue;
                     }
 
+                    bool alreadyHasOverload = method.TryGetAttributeWithType(Model.Compilation.GetTypeByMetadataName("Windows.Foundation.Metadata.OverloadAttribute"), out AttributeData _);
+                    if (alreadyHasOverload)

```

Comment:
I wonder if we should be skipping when there is an attribute or if we should be checking the overloaded method name in the attribute if one is set and then using that below rather than the one we generate.  Once you add the test case, we should be able to see whether the overload attribute is ending up in the winmd or not.